symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/message/util/MessageUtil.java

```diff
@@ -0,0 +1,81 @@
+package com.symphony.bdk.core.service.message.util;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.text.StringEscapeUtils;
+import org.apiguardian.api.API;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Helper class for pre-processing an outgoing message
+ */
+@API(status = API.Status.EXPERIMENTAL)
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public final class MessageUtil {
+  private static Map<String, String> tokens = initializeMap();
+  private static final Pattern PATTERN = initializePattern(tokens);
+
+  /**
+   * This method takes care of all special characters placed within the messageML that must be HTML-escaped
+   * to have a valid MessageML format.
+   *
+   * @param rawText text to be parsed
+   * @return text in a valid messageML format
+   */
+  public static String escapeSpecialChars(String rawText) {
+    Matcher matcher = PATTERN.matcher(rawText);
+    StringBuffer parsedText = new StringBuffer();
```

Comment:
we can use a StringBuilder as we don't need synchronization here

Reply:
here is StringBuffer instead of StringBuilder because `appendReplacement` and `appendTail` require both a StringBuffer in input

Reply:
Ok I did not know :)